src/apps/intel_mp/intel_mp.lua

```diff
@@ -693,6 +716,10 @@ function Intel:pull ()
    if self.run_stats and self.sync_timer() then
       self:sync_stats()
    end
+
+   if self.master and pkts < 1 then
+      self.phy_down_state_alarm:check()
```

Comment:
Bear in mind your alarm if only going to be checked if you receive no packets. I'm thinking of the following scenario:

You run the NIC but receive no packets, then your alarm is checked and it will be raised if the link is down. But then if the link recovers you start receiving packets, however the alarm is not checked again, so it doesn't have a chance to clear.

If the alarm purpose is to report the link state I think it should be checked whether packets were received or not.

Reply:
Also a good point - the idea behind checking if pkts < 1 was to avoid checking the link_status register every breath but obviously that doesn't work to detect the link coming back up again. Will have a think about this...